packages/shylu/shylu_node/tacho/src/Tacho_MatrixMarket.hpp

```diff
@@ -312,6 +321,63 @@ template <typename ValueType> struct MatrixMarket {
     file.unsetf(std::ios::scientific);
     file.precision(prec);
   }
+
+  /// \brief dense vector read
+  template <typename DenseMultiVectorType>
+  static int readDenseVectors(const std::string &filename, DenseMultiVectorType &B, const ordinal_type verbose = 0) {
+
+    std::ifstream file;
+    file.open(filename);
+    if (file.good()) {
+      std::cout << "Read RHS from  " << filename << std::endl;
+    } else {
+      std::cout << "Failed to open the RHS file: " << filename << std::endl;
+      return -1;
+    }
+
+    // reading mm header
+    ordinal_type m = B.extent(0), n = B.extent(1);
+    {
+      std::string header;
+      std::getline(file, header);
+      while (file.good()) {
+        char c = file.peek();
+        if (c == '%' || c == '\n') {
+          file.ignore(256, '\n');
+          continue;
+        }
+        break;
+      }
+      file >> m >> n;
+      if ( m != ordinal_type(B.extent(0))) {
```

Comment:
Is there any reason not to check n as well?

Reply:
For now, number of RHSs is reset based on the input file..